src/components/Tooltip/index.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { ReactElement, RefObject, useRef, useState, useEffect } from 'react'
+import TooltipPortal from './TooltipPortal'
+import { TCoordinates, TPlacement } from './types'
+
+export interface TProps {
+  /**
+  * Вложенный в тултип элемент
+  */
+  children: ReactElement,
+  /**
+  * То, что будет отображаться в тултипе
+  */
+  content: React.ReactNode,
+  /**
+  * Положение тултипа, относительно элемента
+  */
+  placement: TPlacement,
+  /**
+  * Тултип цвета успеха
+  */
+  success?: boolean,
+  /**
+  * Тултип цвета ошибки
+  */
+  error?: boolean,
+}
+
+const getCoordinates = (node: RefObject<HTMLDivElement>): TCoordinates | undefined => {
+  if (node.current !== null) {
+    const rect = node.current.getBoundingClientRect()
+    return {
+      width: rect.width,
+      height: rect.height,
+      left: rect.left + window.pageXOffset,
+      top: rect.top + window.pageYOffset,
+    }
+  }
+}
+
+/**
+ * Компонент для всплывающих при наведении подсказок
+ */
+
+const Tooltip = ({
+  content,
+  placement = 'bottom',
+  success = false,
+  error = false,
+  children,
+}: TProps): JSX.Element => {
+  const childRef = useRef<HTMLDivElement>(null)
+  const [isOpen, setOpenStatus] = useState<boolean>(false)
+  const [coords, setCoords] = useState<TCoordinates | undefined>(undefined)
+
+  const hoverTimeout = useRef<number>()
+
+  useEffect(() => {
+    setCoords(getCoordinates(childRef))
+    return () => window.clearTimeout(hoverTimeout.current)
+  }, [])
+
+  const handleMouseLeave = (): void => {
+    window.clearTimeout(hoverTimeout.current)
```

Comment:
попробуй без window, например мб закастить тип через 
```
setTimeout( ... ) as number
```

если не сработает то забей

Reply:
Conversion of type 'Timeout' to type 'number' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.

ОРЁТ